mqtt/client.py

```diff
@@ -40,34 +41,64 @@ def on_connect(self, client, userdata, flags, rc):
             logging.getLogger('root_logger').info(f'[MQTT]: Successfully connected to broker {self.broker_host}.')
         else:
             raise MQTTException(
-                f'[MQTT]: Connection to broker {self.broker_host} refused. Exited with code {MQTTError(rc).name}'
+                f'[MQTT]: Connection to broker {self.broker_host} refused.',
+                status_code=mqtt_status.MQTT_ERR_CONN_REFUSED
             )
 
     def on_disconnect(self, client, userdata, flags, rc=0):
         if rc == 0:
             logging.getLogger('root_logger').info(f'[MQTT]: Successfully disconnected from broker {self.broker_host}.')
         else:
             raise MQTTException(
-                f'[MQTT]: Disconnection from broker {self.broker_host} refused. Exited with code {MQTTError(rc).name}'
+                f'[MQTT]: Disconnection from broker {self.broker_host} refused.',
+                status_code=mqtt_status.MQTT_ERR_CONN_LOST
             )
 
     def on_message(self, client, userdata, msg):
         topic = msg.topic
         string_message = str(msg.payload.decode('utf-8'))
-        float_message = float(msg.payload.decode('utf-8'))
+        dict_message = json.loads(string_message)
 
         with self.app.app_context():
-            self.socketio.emit(topic,  {'number': float_message}, namespace='/test_web_socket')
-            logging.getLogger('root_logger').info(f'[SocketIO]: Sent message: {string_message} to client.')
-
-            if not topic in self.time:
-                self.time[topic] = 0
-            if self.time.get(topic) >= self.granularity:
-                logging.getLogger('root_logger').info(f'[PostgreSQL]: Created sensor instance: {topic}')
-                Sensor(name=topic, message=string_message).create()
-                self.time[topic] = 0
-        self.time[topic] += 1
-        logging.getLogger('root_logger').info(f'[MQTT]: Message received: {string_message}')
+            module = Module.query.filter_by(mac=topic).first()
+
+            # Request result
+            if dict_message.get('result'):
+
+                if dict_message.get('result') == 'OK':
+                    logging.getLogger('root_logger').info(f'[MQTT]: Message received: {string_message}')
+                elif dict_message.get('result') == 'ERROR':
+                    raise MQTTException(dict_message.get('details'), status_code=mqtt_status.MQTT_ERR_NOT_SUPPORTED)
+                else:
+                    logging.getLogger('root_logger').info(f'[MQTT]: Message received: {string_message}')
+
+            # Value update asynchronously
+            elif dict_message.get('values'):
+                values = dict_message.get('values')
+
+                for item in values:
+                    device = Device.query.get(item.get('id'))
+
+                    if not device:
+                        raise MQTTException(
+                            f'Špecifikovaný device nebol nájdený: {item.get("device_id")}',
```

Comment:
not sure about the `id` and `device_id` keys are present in the JSON message. Please reply if I'm wrong, or else.

Reply:
Yeah you are right about it, but I suggested ( in comment) for format change: https://brewmasters.atlassian.net/wiki/spaces/TD/pages/7274550/MQTT+API

If it can't be changed to that format, I will make proper changes back.

Reply:
thx for your fast reply and clarification, i didnt notice the comment in the DOCS.. approved